testplan/common/utils/match.py

```diff
@@ -400,3 +413,63 @@ def get_between(self, mark1=None, mark2=None):
                 lines_between.append(line)
             separator = b"" if self.binary else ""
             return separator.join(lines_between)
+
+    @contextmanager
+    def expect(
```

Comment:
I would do this the other way around, not the logmatcher know about the existence of a result, but maybe the result knows something about the logmatcher and just use it's api

insteadl of 

width log_matcher expect(resutl,...)

I would do 

result.exepct_logs(logmatcher,...)

here the intent is more apparent, that you do an assertion. Not passing a result object to a supposed to be simple utility class that does log matching.

Reply:
To say the least, I don't want our user to have 2 options doing the same thing, lol.

They used to do driver.log_matcher.match(), it's easier adapt the mind to just add the new functionality as driver.log_matcher.expect_logs() - rather than if you do vanilla log match, do it from driver; else if you need a context manager, do it with result, this complicates things.

I think result can probably passed to the context manager as an optional parameter. Not very mind clean I know...

Reply:
You are right they using driver.log_matcher.match() and then they are doing an assertion with the result if they need it, and in the form and with description as they see fit. So in my view having it in assertion is much closer to current usecase.